src/components/ScheduleSetup.tsx

```diff
@@ -543,67 +598,120 @@ export default class ScheduleSetup extends React.Component<
-          console.log("resource saved:", v);
-        });
-        // update CarePlan with the newly created CommunicationRequests
-        this.state.carePlan.setCommunicationRequests(
-          communicationRequests.filter(
-            (cr) => cr.resourceType === "CommunicationRequest"
-          )
+    this.getCompletedCommunicationRequests().then((bundle: Bundle) => {
+      let completedCRs: CommunicationRequest[] = [];
+      if (bundle.type === "searchset" && bundle.entry) {
+        completedCRs = bundle.entry.map(
+          (e: IBundle_Entry) => {
+            if (e.resource.resourceType === "CommunicationRequest") {
+              return CommunicationRequest.from(
+                e.resource as ICommunicationRequest
+              );
+            } else {
+              return null;
+            }
+          }
         );
-
-        // create resource on server
-        let p;
-        if (this.state.carePlan.id) {
-          p = client.update(this.state.carePlan);
-        } else {
-          p = client.create(this.state.carePlan);
-        }
-        p.then(
-          (savedCarePlan: IResource) => {
-            this.onSaved(savedCarePlan);
-            let updatePromises = communicationRequests.map(
-              (c: CommunicationRequest) => {
-                c.basedOn = [{ reference: `CarePlan/${savedCarePlan.id}` }];
-                return client
-                  .update(c)
-                  .then((value: any) => CommunicationRequest.from(value));
-              }
-            );
-            Promise.all(updatePromises).then(
-              (updatedCommunicationRequests: CommunicationRequest[]) => {
-                updatedCommunicationRequests.forEach(
-                  (v: CommunicationRequest) => {
-                    console.log("CommunicationRequest updated:", v);
-                  }
-                );
-              }
+      }
+      let promises = this.state.carePlan.communicationRequests
+        .filter((c: CommunicationRequest) => {
+          if (c.status !== "active") return true;
+          // for a current active CommunicationRequest
+          // check for matching CommunicationRequest that has been completed, i.e. sent (status === 'completed')
+          const matchedCompletedCr = completedCRs.find(
+            (o: CommunicationRequest) => c?.id && o?.id === c.id
+          );
+          if (matchedCompletedCr) {
+            console.log(
+              "A message has been sent for CommunicationRequest ",
+              c
             );
-          },
-          (reason: any) => this.onRejected(reason)
-        );
-      },
-      this.onRejected
-    );
+            this.setState({
+              saveNotes: [
+                ...(this.state.saveNotes ?? []),
+                `The message dated <b>${new Date(c.occurrenceDateTime).toLocaleString()}</b> -- ${c.getText()} -- had been sent by the system while you were editing it.<br/><i>This message is not saved</i>.`,
```

Comment:
I'd suggest refining this to make it a bit more legible:
```
                `The message dated <b>${new Date(c.occurrenceDateTime).toLocaleString()}</b> -- ${c.getText()} -- had been sent by the system while you were editing it.<br/><i>This message is not saved</i>.`,

                `<i>You can no longer edit the following message, as it was sent by the system on <b>${new Date(c.occurrenceDateTime).toLocaleString()}</b></i>:<br/>
${c.getText()}`,
```

Reply:
thank you @mcjustin I have updated the text.  Please see the updated  screenshot in the description.